plugins/uefi-ssp-policy/fu-uefi-ssp-policy-plugin.c

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2024 Richard hughes <Richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#include "config.h"
+
+#include "fu-uefi-ssp-policy-plugin.h"
+
+struct _FuUefiSspPolicyPlugin {
+	FuPlugin parent_instance;
+};
+
+G_DEFINE_TYPE(FuUefiSspPolicyPlugin, fu_uefi_ssp_policy_plugin, FU_TYPE_PLUGIN)
+
+typedef enum {
+	SHIM_SSP_POLICY_LATEST = 1,
+	SHIM_SSP_POLICY_AUTOMATIC = 2,
+	SHIM_SSP_POLICY_DELETE = 3,
+} ShimSspPolicy;
+
+static gboolean
+fu_uefi_ssp_policy_plugin_bootmgr_found(FuPlugin *plugin)
+{
+	FuContext *ctx = fu_plugin_get_context(plugin);
+	g_autoptr(GPtrArray) esp_files = NULL;
+	g_autoptr(GError) error_local = NULL;
+
+	esp_files = fu_context_get_esp_files(ctx,
+					     FU_CONTEXT_ESP_FILE_FLAG_INCLUDE_FIRST_STAGE,
+					     &error_local);
+	if (esp_files == NULL) {
+		g_warning("failed to get files on ESP: %s", error_local->message);
+		return FALSE;
+	}
+	for (guint i = 0; i < esp_files->len; i++) {
+		FuFirmware *esp_file = g_ptr_array_index(esp_files, i);
+		const gchar *fn = fu_firmware_get_filename(esp_file);
+		if (fn != NULL && g_strstr_len(fn, -1, "bootmgfw") != NULL)
+			return TRUE;
+	}
+	return FALSE;
+}
+
+static void
+fu_uefi_ssp_policy_plugin_add_security_attrs(FuPlugin *plugin, FuSecurityAttrs *attrs)
+{
+	FuContext *ctx = fu_plugin_get_context(plugin);
+	FuEfivars *efivars = fu_context_get_efivars(ctx);
+	g_autoptr(FwupdSecurityAttr) attr = NULL;
+
+	if (fu_plugin_has_flag(plugin, FWUPD_PLUGIN_FLAG_DISABLED))
+		return;
+
+	/* create attr */
+	attr = fu_plugin_security_attr_new(plugin, FWUPD_SECURITY_ATTR_ID_SSP_POLICY_VARS);
+	fwupd_security_attr_add_flag(attr, FWUPD_SECURITY_ATTR_FLAG_ACTION_CONFIG_OS);
+	fwupd_security_attr_add_flag(attr, FWUPD_SECURITY_ATTR_FLAG_RUNTIME_ISSUE);
+	fwupd_security_attr_set_result_success(attr, FWUPD_SECURITY_ATTR_RESULT_FOUND);
+	fu_security_attrs_append(attrs, attr);
+
+	/* either key is not found */
+	if (!fu_efivars_exists(efivars, FU_EFI_SIGNATURE_GUID_MICROSOFT, "SkuSiPolicyVersion") ||
+	    !fu_efivars_exists(efivars, FU_EFI_SIGNATURE_GUID_MICROSOFT, "SkuSiPolicyVersion")) {
+		/* is bootmgr listed in BootOrder? */
+		if (fu_uefi_ssp_policy_plugin_bootmgr_found(plugin)) {
+			fwupd_security_attr_set_result(attr, FWUPD_SECURITY_ATTR_RESULT_NOT_VALID);
+			return;
+		}
+
+		/* we can fix this! */
+		fwupd_security_attr_add_flag(attr, FWUPD_SECURITY_ATTR_FLAG_CAN_FIX);
```

Comment:
I think you need to check whether `SSPPolicy` already exists though don't you?  It can only be fixed if the file doesn't exist.